### ISSUE TITLE
Docker: ksql-test-runner is missing hamcrest dependency in 0.7.0

### ISSUE THREAD

**Issue by colinhicks** [confluentinc/ksql#4584]
**Describe the bug**
Using the ` confluentinc/ksqldb-cli:0.7.0` Docker image, `ksql-test-runner` throws `java.lang.NoClassDefFoundError: org/hamcrest/Matcher`

**To Reproduce**
(Note: this is a minimal reproduction. The test case here isn't actually valid; it's just enough to initialize the runner so that the issue can be surfaced, without requiring other services.)
In some scratch directory:
1. `echo "create stream astream (x integer) with (kafka_topic='atopic', value_format='json');" > statements.sql`
2. `echo '{"outputs":[{"topic": "btopic", "value": 0}]}' > output.json`
3. `echo '{"inputs":[{"topic": "atopic", "value": 0}]}' > input.json`
4. `docker run -v $(pwd):/test confluentinc/ksqldb-cli:0.7.0 ksql-test-runner -s test/statements.sql -i test/input.json -o test/output.json`

**Expected behavior**
The test runner should not hit the exception below.

**Actual behaviour**
```
<snip INFO logging ...>
Exception in thread "main" java.lang.BootstrapMethodError: java.lang.NoClassDefFoundError: org/hamcrest/Matcher
	at io.confluent.ksql.test.tools.TestCaseBuilder.createTest(TestCaseBuilder.java:76)
	at io.confluent.ksql.test.tools.TestCaseBuilder.lambda$buildTests$0(TestCaseBuilder.java:50)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.stream.Streams$StreamBuilderImpl.forEachRemaining(Streams.java:419)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at io.confluent.ksql.test.tools.TestCaseBuilder.buildTests(TestCaseBuilder.java:51)
	at io.confluent.ksql.test.tools.KsqlTestingTool.runWithTripleFiles(KsqlTestingTool.java:127)
	at io.confluent.ksql.test.tools.KsqlTestingTool.main(KsqlTestingTool.java:55)
Caused by: java.lang.NoClassDefFoundError: org/hamcrest/Matcher
	... 12 more
Caused by: java.lang.ClassNotFoundException: org.hamcrest.Matcher
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 12 more
```

**Comment by colinhicks** [comment]
fixed in b023cd0

**Comment by colinhicks** [comment]
Re-opening to indicate the need to cherry-pick this onto the 0.7.1 branch.
